### PR TITLE
Add conceptual content for partial properties

### LINKED ISSUE #27701: The description for "Accessibility" can have a few more details.
Under this section its mentioned that "The default accessibility is private." I think it can be further mentioned here that its internal for the Class members. Because for Class the default accessibility is "Internal". 
Kindly confirm.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2eb21e6e-7798-e557-be3f-f27ab9c93e05
* Version Independent ID: a526a9cc-938e-7528-b5b4-4879c5085152
* Content: [Classes, structs, and records](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/object-oriented/)
* Content Source: [docs/csharp/fundamentals/object-oriented/index.md](https://github.com/dotnet/docs/blob/main/docs/csharp/fundamentals/object-oriented/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

---
[Associated WorkItem - 297236](https://dev.azure.com/msft-skilling/Content/_workitems/edit/297236)

### LINKED ISSUE #29962: Record or Class or Struct? Decision Tree
I was certain that this page would help me decide between records and structs and classes. It's an unexpectedly unclear decision process that I think new developers struggle to understand. Why not provide a decision tree? A decision tree like (is it this? yes/no). If such a tree comes across as too simple, then it will be simple to create, I hope. But the struggle is real. Especially with students. 

Some starter questions:

1. Is my data mutable?
2. Do you want value-based equality?
3. Do you want non-destructive mutation?
4. Do you want a reference type?

Also see: https://twitter.com/jerrynixon/status/1392721462423953410
Also see: https://twitter.com/jerrynixon/status/1539605996481429505

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2eb21e6e-7798-e557-be3f-f27ab9c93e05
* Version Independent ID: a526a9cc-938e-7528-b5b4-4879c5085152
* Content: [Classes, structs, and records in C#](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/object-oriented/)
* Content Source: [docs/csharp/fundamentals/object-oriented/index.md](https://github.com/dotnet/docs/blob/main/docs/csharp/fundamentals/object-oriented/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

---
[Associated WorkItem - 297237](https://dev.azure.com/msft-skilling/Content/_workitems/edit/297237)

### LINKED ISSUE #30421: Combine with programming guide content
This article provides similar information to the [Properties](https://docs.microsoft.com/dotnet/csharp/programming-guide/classes-and-structs/properties) section of the programming guide. There's also a lot of redundancy. This article is newer, and should be the main source for an update. Any concepts in the programming guide article could be included here, or ported to a new article.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 1e614217-286d-2231-f0c7-d37e5a31b2d5
* Version Independent ID: 070d932f-bc38-a766-37cc-ec861e74be1c
* Content: [Properties in C#](https://docs.microsoft.com/en-us/dotnet/csharp/properties)
* Content Source: [docs/csharp/properties.md](https://github.com/dotnet/docs/blob/main/docs/csharp/properties.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

---
[Associated WorkItem - 297235](https://dev.azure.com/msft-skilling/Content/_workitems/edit/297235)

### LINKED ISSUE #35166: Include examples with the `required` keyword
See https://github.com/dotnet/docs/pull/35123#discussion_r1177967630

This article could use an update for C# 11 to include examples using the `required` keyword.

It might be best to make those changes in the C# 12 timeframe, when the `field` keyword may be added.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 1e614217-286d-2231-f0c7-d37e5a31b2d5
* Version Independent ID: 070d932f-bc38-a766-37cc-ec861e74be1c
* Content: [Properties in C#](https://learn.microsoft.com/en-us/dotnet/csharp/properties)
* Content Source: [docs/csharp/properties.md](https://github.com/dotnet/docs/blob/main/docs/csharp/properties.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

---
[Associated WorkItem - 297234](https://dev.azure.com/msft-skilling/Content/_workitems/edit/297234)

### LINKED ISSUE #40150: Add a table for all property access modifiers
### Type of issue

Missing information

### Description

We should add a table that explains the relationships and behavior for different access modifiers: `get`, `set`, `init`, and `required`. It should also explain read-only auto-implemented properties, where the property can be set in a constructor, even though no `set` accessor is declared.

It should also have the common cases for adding an access modifier on the `set` accessor.

### Page URL

https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/csharp/programming-guide/classes-and-structs/properties.md

### Document Version Independent Id

05423053-63de-3e56-0929-2ae477d577df

### Article author

@BillWagner

### Metadata

* ID: 8fb1d95e-a652-a965-f817-d49554a07388 
* Service: **dotnet-csharp**
* Sub-service: **fundamentals**

---
[Associated WorkItem - 297644](https://dev.azure.com/msft-skilling/Content/_workitems/edit/297644)

### LINKED ISSUE #42135: Combine this article with the programming guide article on indexers
### Type of issue

Typo

### Description

Similar to #30421 

This article contains considerable duplication to the indexers article in the programming guide. Combine them.

### Page URL

https://learn.microsoft.com/en-us/dotnet/csharp/indexers

### Content source URL

https://github.com/dotnet/docs/blob/main/docs/csharp/indexers.md

### Document Version Independent Id

4f2c312e-609c-a643-4af6-c12c6cf74738

### Article author

@BillWagner

### Metadata

* ID: bcc06ead-fe51-85db-1340-87f6473db769 
* Service: **dotnet-csharp**
* Sub-service: **fundamentals**

---
[Associated WorkItem - 297233](https://dev.azure.com/msft-skilling/Content/_workitems/edit/297233)

### PR DESCRIPTION
**Pull request by BillWagner**
Contributes to #41572 

- Rename "partial-methods" as "partial-members"
- Publish partial properties speclet.
- Add information on partial properties and indexers to the language reference article on partial types and partial members.
- Add links or notes on partial members in the fundamentals section. Fix open issues on affected articles:
  - Fixes #27701: Explain default accessibility on types and members.
  - Fixes #29962: Add text on deciding between `struct`, `class`, and `record` types.
- Combine duplicated articles on properties and indexers
  - Fixes #30421
  - Fixes #42135 
- Major edit pass on the duplicated Properties article.
  - Fixes #35166 - Add an example with `required` properties.
  - Fixes #40150 - I didn't add a table here. Instead, I added text to explain the rules. I tried a table, and it really looked ugly.
- Major edit pass on the duplicated Indexers article.
- Check related articles for links, quick edit pass
- Update all links.
- Add note and links in "What's new in C# 13"

All lot of the updates are small. It may be easier to review commit-by-commit.
 

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/advanced-topics/expression-trees/index.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/advanced-topics/expression-trees/index.md) | [docs/csharp/advanced-topics/expression-trees/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/expression-trees/index?branch=pr-en-us-42177) |
| [docs/csharp/advanced-topics/reflection-and-attributes/index.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/advanced-topics/reflection-and-attributes/index.md) | [docs/csharp/advanced-topics/reflection-and-attributes/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/reflection-and-attributes/index?branch=pr-en-us-42177) |
| [docs/csharp/event-pattern.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/event-pattern.md) | [docs/csharp/event-pattern](https://review.learn.microsoft.com/en-us/dotnet/csharp/event-pattern?branch=pr-en-us-42177) |
| [docs/csharp/fundamentals/exceptions/how-to-handle-an-exception-using-try-catch.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/fundamentals/exceptions/how-to-handle-an-exception-using-try-catch.md) | [docs/csharp/fundamentals/exceptions/how-to-handle-an-exception-using-try-catch](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/how-to-handle-an-exception-using-try-catch?branch=pr-en-us-42177) |
| [docs/csharp/fundamentals/types/classes.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/fundamentals/types/classes.md) | [docs/csharp/fundamentals/types/classes](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/classes?branch=pr-en-us-42177) |
| [docs/csharp/fundamentals/types/index.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/fundamentals/types/index.md) | [docs/csharp/fundamentals/types/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/index?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/compiler-messages/cs8795.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/compiler-messages/cs8795.md) | [docs/csharp/language-reference/compiler-messages/cs8795](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs8795?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/compiler-messages/expression-tree-restrictions.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/compiler-messages/expression-tree-restrictions.md) | [docs/csharp/language-reference/compiler-messages/expression-tree-restrictions](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/expression-tree-restrictions?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/compiler-messages/inline-array-errors.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/compiler-messages/inline-array-errors.md) | [docs/csharp/language-reference/compiler-messages/inline-array-errors](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/inline-array-errors?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/keywords/from-clause.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/keywords/from-clause.md) | [docs/csharp/language-reference/keywords/from-clause](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/from-clause?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/keywords/get.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/keywords/get.md) | [docs/csharp/language-reference/keywords/get](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/get?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/keywords/index.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/keywords/index.md) | [docs/csharp/language-reference/keywords/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/index?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/keywords/partial-type.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/keywords/partial-type.md) | [docs/csharp/language-reference/keywords/partial-type](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/partial-type?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/keywords/select-clause.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/keywords/select-clause.md) | [docs/csharp/language-reference/keywords/select-clause](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/select-clause?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/keywords/set.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/keywords/set.md) | [docs/csharp/language-reference/keywords/set](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/set?branch=pr-en-us-42177) |
| [docs/csharp/language-reference/xmldoc/recommended-tags.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/language-reference/xmldoc/recommended-tags.md) | [docs/csharp/language-reference/xmldoc/recommended-tags](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags?branch=pr-en-us-42177) |
| [docs/csharp/misc/cs0501.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/misc/cs0501.md) | [docs/csharp/misc/cs0501](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs0501?branch=pr-en-us-42177) |
| [docs/csharp/misc/cs0751.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/misc/cs0751.md) | [docs/csharp/misc/cs0751](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/cs0751?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/classes-and-structs/access-modifiers.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/classes-and-structs/access-modifiers.md) | [docs/csharp/programming-guide/classes-and-structs/access-modifiers](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/classes-and-structs/auto-implemented-properties.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/classes-and-structs/auto-implemented-properties.md) | [docs/csharp/programming-guide/classes-and-structs/auto-implemented-properties](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/auto-implemented-properties?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/classes-and-structs/interface-properties.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/classes-and-structs/interface-properties.md) | [docs/csharp/programming-guide/classes-and-structs/interface-properties](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/interface-properties?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods.md) | [docs/csharp/programming-guide/classes-and-structs/partial-classes-and-methods](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/partial-classes-and-methods?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/classes-and-structs/properties.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/classes-and-structs/properties.md) | [docs/csharp/programming-guide/classes-and-structs/properties](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/classes-and-structs/using-properties.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/classes-and-structs/using-properties.md) | [docs/csharp/programming-guide/classes-and-structs/using-properties](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/using-properties?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/indexers/index.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/indexers/index.md) | [docs/csharp/programming-guide/indexers/index](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/index?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/indexers/indexers-in-interfaces.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/indexers/indexers-in-interfaces.md) | [docs/csharp/programming-guide/indexers/indexers-in-interfaces](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/indexers-in-interfaces?branch=pr-en-us-42177) |
| [docs/csharp/programming-guide/indexers/using-indexers.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/programming-guide/indexers/using-indexers.md) | [docs/csharp/programming-guide/indexers/using-indexers](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/indexers/using-indexers?branch=pr-en-us-42177) |
| [docs/csharp/tour-of-csharp/tips-for-java-developers.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/tour-of-csharp/tips-for-java-developers.md) | [docs/csharp/tour-of-csharp/tips-for-java-developers](https://review.learn.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tips-for-java-developers?branch=pr-en-us-42177) |
| [docs/csharp/whats-new/csharp-13.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/whats-new/csharp-13.md) | [docs/csharp/whats-new/csharp-13](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-13?branch=pr-en-us-42177) |
| [docs/csharp/whats-new/csharp-version-history.md](https://github.com/dotnet/docs/blob/cd95b5b17e9afc0af1a0c12e040e0b0e328f56ab/docs/csharp/whats-new/csharp-version-history.md) | [docs/csharp/whats-new/csharp-version-history](https://review.learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-version-history?branch=pr-en-us-42177) |

</details>

> [!NOTE]
> This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/docs/pull/42177/checks).

### CODE REVIEW

**Comment by BillWagner** [comment]
The snippets 5000 error is a false positive. This PR moves samples from locations without a project to a location that has a project to compile.

File: docs/csharp/language-reference/keywords/partial-member.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+title: "Partial members"
+ms.date: 08/15/2024
+f1_keywords: 
+  - "partialmethod_CSharpKeyword"
+helpviewer_keywords: 
+  - "partial methods [C#]"
+---
+# Partial member (C# Reference)
+
+A partial member has one *declaring declaration* and often one *implementing declaration*. The *declaring declaration* doesn't include a body. The *implementing declaration* provides the body of the member. Partial members enable class designers to provide member hooks, similar to event handlers, that developers can decide to implement or not. If the developer doesn't supply an implementing declaration, the compiler removes the declaring declaration at compile time. The following conditions apply to partial members:
```
Comment (IEvangelist) [inline docs/csharp/language-reference/keywords/partial-member.md:12]: That reads a lot like the functionality of the `abstract` keyword, should we call attention to how this differs from that?

**Review by IEvangelist — APPROVED** [review]
LGTM, just had one suggestion.

File: docs/csharp/language-reference/keywords/partial-member.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/language-reference/keywords/partial-member.md:12]: I updated this to emphasize the tools aspect of `partial`.

**Pull request merged**